src/Containers/HorizontalBarChart/HorizontalBarChart.tsx

```diff
@@ -0,0 +1,176 @@
+import React,{useCallback, useLayoutEffect, useRef,useState,useMemo} from 'react'
+import {BarChart,XAxis,YAxis,Bar,Text,Cell} from 'recharts'
+import styled from 'styled-components'
+import Theme from '@Themes/ThemeTemplate'
+import {MainTheme} from '@Themes'
+import {Mixins} from '@Utils'
+import {Paragraph} from '@Text'
+
+const FLEX_EQUAL_SPACE=1
+const SUMMARY_BACKGROUND_COLOR='#d3d3d3'
+const BAR_LABEL_LEFT_MARGIN=5
+const SUMMARY_CONTAINER_MARGIN=14
+const IS_TOTAL_COLOR='#d3d3d3'
+const GREEN_COLOR_MARGIN=5
+const ORANGE_COLOR_MARGIN=15
+const BAR_SIZE=20
+
+interface IDataItem{
+    label:string;
+    value:number;
+    isTotal:boolean;
+}
+
+interface IChartProperties{
+    margin?:object;
+    data:IDataItem[];
+    unit:string;
+}
+
+export interface IHorizontalBarChartProps extends IChartProperties{
+    header:string;
+    summaryHeader:string;
+    summaryDescription:string;
+    width?:number;
+}
+
+export const HorizontalBarChart:React.FC<IHorizontalBarChartProps>=({data,header,summaryHeader,summaryDescription,margin,unit,...props}):React.ReactElement=>{
+    const [chartWidth,setChartWidth]=useState<number>()
+    const [chartHeight,setChartHeight]=useState<number>()
+
+    // used to compute width and height for the chart based on width and height of summary container
+    const summaryContainerRef=useRef<HTMLDivElement>(null)
+
+    /**
+     * compute dynamically the width and height of the chart
+     */
+    useLayoutEffect(()=>{
+        setChartHeight(summaryContainerRef.current?.clientHeight)
+        setChartWidth(summaryContainerRef.current?.clientWidth)
+    },[])
+
+    /**
+     * this is the customized bar label used to tell the value of the bars
+     */
+    const BarLabel=useCallback(({value,x,y,width:barWidth,height:barHeight}:any):React.ReactElement<SVGElement>=>{
+        /**
+         * this computes the x value for the text, based on x value of the bar, its width (bar width) and a little margin
+         * @returns the x value for the text
+         */
+        const getX=()=>x+barWidth+BAR_LABEL_LEFT_MARGIN
+
+        /**
+         * this computes the y value for the text, based on y value for the bar, and bar height formula to place it in the center
+         * of the bar
+         * @returns the y value for the text
+         */
+        const getY=()=>y+barHeight/2+barHeight/4
```

Comment:
Unclear as to why you put 2 and 4.

2: Center Text in Div
4: 

Function Name is unclear
const getCenterOfBarYValue = () => ;